src/StructuredLogger.Tests/BinaryLogger/BinLogReaderTests.cs

```diff
@@ -0,0 +1,319 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Logging.StructuredLogger;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Build.Logging.StructuredLogger.UnitTests
+{
+    /// <summary>
+    /// A fake Progress implementation to capture progress reports during tests.
+    /// </summary>
+    public class FakeProgress : Progress<ProgressUpdate>, Progress

```

Comment:
Again - 'hand-mocking' went wrong - it mixes System.Progress StructuredLogger.Progress